executor/src/witgen/jit/single_step_processor.rs

```diff
@@ -164,9 +228,55 @@ mod test {
         let err = generate_single_step(input, "M").err().unwrap();
         assert_eq!(
             err.to_string(),
-            "Unable to derive algorithm to compute values for witness columns in the next row for the following columns:\n\
+            "Unable to derive algorithm to compute values for witness columns in the next row and\n\
+            unable to branch on a variable. The following columns are still missing:\n\
             M::Y\n\
-            and 0 identities are missing."
+            The following identities have not been fully processed:\n"
+        );
+    }
+
+    #[test]
+    fn branching() {
+        let input = "
+    namespace VM(256);
+        let A: col;
+        let B: col;
+        let instr_add: col;
+        let instr_mul: col;
+        let pc: col;
+
+        col fixed LINE = [0, 1] + [2]*;
+        col fixed INSTR_ADD = [0, 1] +  [0]*;
+        col fixed INSTR_MUL = [1, 0] + [1]*;
+
+        pc' = pc + 1;
+        [ pc, instr_add, instr_mul ] in [ LINE, INSTR_ADD, INSTR_MUL ];
+
+        instr_add * (A' - (A + B)) + instr_mul * (A' - A * B)  + (1 - instr_add - instr_mul) * (A' - A) = 0;
```

Comment:
```suggestion
        instr_add * (A' - (A + B)) + instr_mul * (A' - A * B) + (1 - instr_add - instr_mul) * (A' - A) = 0;
```

Reply:
In this case, the `(1 - instr_add - instr_mul) * (A' - A)` is not needed, because one instruction will always be active. But removing that leads to a failure, because the processor doesn't know that.

Not sure if that's a problem in practice, but is this a limitation we can fix eventually?

Reply:
I think the range constraints from the fixed lookup should fix that.